lib/auth.ml

```diff
@@ -27,6 +27,20 @@ type state =
   | Inprogress of (string * string * int)
   | Done
 
+type pubkeyauth = {
+  pubkey : Hostkey.pub ;
+  session_id : string ;
+  service : string ;
+  sig_alg : Hostkey.alg ;
+  signed : string ;
+}
+
+let pubkey_of_pubkeyauth { pubkey; _ } = pubkey
+
+type userauth =
+  | Password of string
+  | Pubkey of pubkeyauth
+
```

Comment:
Moving this into Auth made sense to me as the related verification code lives there. But it makes it a bit more annoying to make pubkeyauth opaque as we need to construct it in Server.

Of course, we can create a awa.mli and better control what is exposed to library users.

Reply:
After some thought: Maybe this is moved back to Server.ml, and the `Auth.user`/`Auth.db` related code is moved into Awa_mirage?

Reply:
I'd be happy to have a awa.mli :) about code moving, I'm not sure, if you think it is worth, please do it :) though moving purely functional stuff into the effectful mirage layer is something I'd avoid.

Reply:
Another motivation for me of moving that code to `Awa_mirage` is I may eventually want to get rid of that code tbh. After all, a (static) list of users in memory is maybe not the best fit for all. We may want to use various password hashing algorithms. Or we may accept public keys based on their fingerprint (this may be a very bad idea; I don't know).